scipy/stats/_continuous_distns.py

```diff
@@ -9033,6 +9034,101 @@ def _munp(self, n, b):
 
 rice = rice_gen(a=0.0, name="rice")
 
+class irwinhall_gen(rv_continuous):
+    r"""An Irwin-Hall (Uniform Sum) continuous random variable.
+
+    An `Irwin-Hall <https://en.wikipedia.org/wiki/Irwin-Hall_distribution/>`_ 
+    [1]_ [2]_ continuous random variable is the sum 
+    of :math`n` independent standard uniform r.v.'s. 
+    
+    %(before_notes)s
+
+    Notes
+    -----
+    Applications include `Rao's Spacing Test <https://jammalam.faculty.pstat.ucsb.edu/html/favorite/test.htm>`_, 
+    a more powerful alternative to the Rayleigh test 
+    when the data are not unimodal, and radar [3]_.
+
+    Conveniently, the pdf and cdf are the :math:`n`-fold convolution of 
+    the ones for the standard uniform distribution,
+    which are well known to be just cardinal B-splines of degree :math:`n-1` 
+    with knots evenly spaced from :math:`1` to :math:`n`.
+
+    The Bates distribution is simply the Irwin-Hall distribution scaled 
+    by :math:`1/n`: ``bates = irwinhall(n, name="bates", scale=1/n)``
+
+    
+    %(after_notes)s
+
+    References
+    ----------
+    .. [1] HALL PHILIP, "THE DISTRIBUTION OF MEANS FOR SAMPLES OF SIZE N DRAWN 
+            FROM A POPULATION IN WHICH THE VARIATE TAKES VALUES BETWEEN 0 AND 1, 
+            ALL SUCH VALUES BEING EQUALLY PROBABLE", 
+            Biometrika, Volume 19, Issue 3-4, December 1927, Pages 240-244, 
+            https://doi.org/10.1093/biomet/19.3-4.240
+    .. [2] J. O. IRWIN, "ON THE FREQUENCY DISTRIBUTION OF THE MEANS OF SAMPLES 
+            FROM A POPULATION HAVING ANY LAW OF FREQUENCY WITH FINITE MOMENTS, 
+            WITH SPECIAL REFERENCE TO PEARSON'S TYPE II", 
+            Biometrika, Volume 19, Issue 3-4, December 1927, Pages 225-239, 
+            https://doi.org/10.1093/biomet/19.3-4.225
+    .. [3] K. Buchanan, T. Adeyemi, C. Flores-Molina, S. Wheeland and D. Overturf, 
+            "Sidelobe behavior and bandwidth characteristics 
+            of distributed antenna arrays," 
+            2018 United States National Committee of 
+            URSI National Radio Science Meeting (USNC-URSI NRSM), 
+            Boulder, CO, USA, 2018, pp. 1-2.
+            https://www.usnc-ursi-archive.org/nrsm/2018/papers/B15-9.pdf
+
+    %(example)s
+    """
+
+    def _argcheck(self, n):
+        return (n > 0) & _isintegral(n) & np.isrealobj(n)
+    
+    def _get_support(self, n):
+        return 0, n 
+    
+    def _shape_info(self):
+        return [_ShapeInfo("n", True, (1, np.inf), (True, False))]
+    
+    def _munp(self, n, t):
+        return (sc.expm1(t)/t)**n
+        
+    @staticmethod
+    def _cardbspl(n):
+        t = np.arange(n+1) 
+        return BSpline.basis_element(t)
```

Comment:
Same here - so that I can confirm, please add a reference and page number.

Reply:
Please leave these unresolved; I'll resolve them when I've confirmed that they have been addressed.

How has this been addressed? I would appreciate a comment right here in the code pointing to the reference number. As a reviewer, I need to do due diligence and check out the source of all this stuff. We all have different backgrounds, and I am not intimitely familiar with this distribution or B-splines, so the connection is not immediately obvious. I can also look it up, but this takes more time, and of course there will be others who inspect this code in the future who would need to do the same. Let's make it easier for everyone - except you, sorry : )

Reply:
@ev-br @steppi I could dive into B-splines and such, but I wonder if you'd be willing to review this part - that the PDF of the Irwin-Hall distribution PDF is as simple as this, no gotchas that you see?

Reply:
@mdhaber take a look at the doc references 4 and 5: from https://www.chebfun.org/examples/approx/BSplineConv.html (by @trefethen and chebfun/chebfun )


>B-splines were introduced by Schoenberg and became a standard method for numerical computation following the work of de Boor [1] and Cox [2] in 1972. In general there is no need for the breakpoints to be uniformly spaced. With uniformly spaced points as in our example, however, the shape approaches a Gaussian as n→∞. One could derive this fact by noting that in Fourier space, each convolution is a multiplication by the Fourier transform of B0, namely a sinc function. This is the same computation that arises in the proof of the Central Limit Theorem, and in fact, the B-splines have a probabilistic interpretation: if B0 represents a random variable with uniform distribution on the interval [−1/2,1/2] then B1 is the distribution for the sum of two independent samples from B0, B2 is the distribution for the sum of three samples from B0, and so on.

Reply:
Another derivation comes from the cumulants of the uniform. Since the IH is just a sum of `n` standard uniforms, you take `n` times the uniform cumulant.

The nth cumulant of U(0,1) is $\frac{B_n}{n}$ where $B_n$ is nth [Bernoulli number](https://en.wikipedia.org/wiki/Bernoulli_number) so the nth cumulant of an IH(m) distribution is $$\frac{mB_n}{n}$$

That immediately implies the odd order cumulants are 0 for $n \geq 3$. We could use `scipy.special.bernoulli` for the higher order ones. 

The formulas for moments in terms of cumulants quickly get very complicated, but it could be done with `sympy`

Reply:
Yeah, that last part is good. Their BSplines are defined on `[−1/2,1/2]` and ours are on `[0, 1]`, I guess?

Reply:
Yes but it doesn't matter due to translation invariance (fortunately) -- basically the equivalent of setting `loc=1/2`

Reply:
@mdhaber, this looks right to me. I'll sign off on this. The reference @nimish provided looks good too.